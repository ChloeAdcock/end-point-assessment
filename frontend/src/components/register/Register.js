import React, { useState, useEffect } from "react";
import Typography from "@material-ui/core/Typography";
import Textfield from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Alert from "@material-ui/lab/Alert";
import Grid from "@material-ui/core/Grid";
import IconButton from "@material-ui/core/IconButton";
import Tooltip from "@material-ui/core/Tooltip";
import Link from "@material-ui/core/Link";
import HelpOutlineIcon from "@material-ui/icons/HelpOutline";
import { Redirect } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { register, closeAlert } from "../../redux/actions/accounts/accounts";

function Register() {
  const dispatch = useDispatch();
  const [username, setUsername] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [fieldError, setFieldError] = useState({
    username: false,
    email: false,
    password: false,
    confirmPassword: false,
  });
  const [fieldTouched, setFieldTouched] = useState({
    username: false,
    email: false,
    password: false,
    confirmPassword: false,
  });
  const errorState = useSelector((state) => state.accounts.registerError);
  const user = useSelector((state) => state.accounts.currentUser);

  const validateEmail = () => {
    // Letters then @ then letters then full stop then letters
    if (!email.match(/^[^@]+@[^@]+\.[^@]+$/)) {
      setFieldError({
        ...fieldError,
        ...{ email: true },
      });
    } else {
      setFieldError({
        ...fieldError,
        ...{ email: false },
      });
    }
  };
  const validatePassword = () => {
    // Alphanumeric, must have a number and a letter, 8-15 characters
    if (!password.match(/^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,15}$/)) {
      setFieldError({
        ...fieldError,
        ...{ password: true },
      });
    } else {
      setFieldError({
        ...fieldError,
        ...{ password: false },
      });
    }
  };

  const validateUsername = () => {
    // Alphanumeric between 5 and 10 characters
    if (!username.match(/^[a-zA-Z0-9]{5,10}$/)) {
      setFieldError({
        ...fieldError,
        ...{ username: true },
      });
    } else {
      setFieldError({
        ...fieldError,
        ...{ username: false },
      });
    }
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (Object.values(fieldError).indexOf(true) === -1) {
      dispatch(
        register({
          username: username,
          email: email,
          password: password,
        })
      );
    }
  };

  useEffect(() => {
    if (password !== confirmPassword) {
      setFieldError({
        ...fieldError,
        ...{ confirmPassword: true },
      });
    } else {
      setFieldError({
        ...fieldError,
        ...{ confirmPassword: false },
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [confirmPassword]);

  const handleClose = () => {
    dispatch(closeAlert());
  };

  if (!errorState && user) {
    return <Redirect to="/home" />;
  } else {
    return (
      <div>
        {errorState && (
          <Alert severity="error" onClose={handleClose}>
            An error has occurred - please try again later
          </Alert>
        )}
        <form onSubmit={handleSubmit}>
          <Typography varient="h1">Register</Typography>
          <div>
            <Grid container spacing={1} alignItems="flex-end">
              <Grid item>
                <Tooltip title="Username must be alphanumeric and 5-10 characters">
                  <IconButton aria-label="info">
                    <HelpOutlineIcon />
                  </IconButton>
                </Tooltip>
              </Grid>
              <Grid item>
                <Textfield
                  label="Username"
                  value={username}
                  onBlur={() => {
                    setFieldTouched({
                      ...fieldTouched,
                      ...{ username: true },
                    });
                  }}
                  error={fieldTouched.username && fieldError.username}
                  onChange={(e) => {
                    setUsername(e.target.value);
                    validateUsername();
                  }}
                  variant="outlined"
                  required
                  helperText={
                    fieldTouched.username &&
                    fieldError.username &&
                    "Invalid username"
                  }
                />
              </Grid>
            </Grid>
          </div>
          <Textfield
            label="Email"
            error={fieldTouched.email && fieldError.email}
            onBlur={() => {
              setFieldTouched({
                ...fieldTouched,
                ...{ email: true },
              });
            }}
            value={email}
            variant="outlined"
            onChange={(e) => {
              setEmail(e.target.value);
              validateEmail();
            }}
            type="email"
            required
            helperText={
              fieldTouched.email && fieldError.email && "Invalid email"
            }
          />
          <div>
            <Grid container spacing={1} alignItems="flex-end">
              <Grid item>
                <Tooltip title="Password must be alphanumeric, 5-10 characters and contain at least one letter and one number">
                  <IconButton aria-label="info">
                    <HelpOutlineIcon />
                  </IconButton>
                </Tooltip>
              </Grid>
              <Grid item>
                <Textfield
                  label="Password"
                  type="password"
                  error={fieldTouched.password && fieldError.password}
                  onBlur={() => {
                    setFieldTouched({
                      ...fieldTouched,
                      ...{ password: true },
                    });
                  }}
                  value={password}
                  onChange={(e) => {
                    setPassword(e.target.value);
                    validatePassword();
                  }}
                  variant="outlined"
                  required
                  helperText={
                    fieldTouched.password &&
                    fieldError.password &&
                    "Invalid password"
                  }
                />
              </Grid>
            </Grid>
          </div>
          <Textfield
            label="Confirm password"
            type="password"
            value={confirmPassword}
            onChange={(e) => setConfirmPassword(e.target.value)}
            error={fieldTouched.confirmPassword && fieldError.confirmPassword}
            onBlur={() => {
              setFieldTouched({
                ...fieldTouched,
                ...{ confirmPassword: true },
              });
            }}
            variant="outlined"
            required
            helperText={
              fieldTouched.confirmPassword &&
              fieldError.confirmPassword &&
              "Passwords do not match"
            }
          />
          <Button type="submit">Submit</Button>
          <Typography>
            <Link href="/login" onClick={(e) => e.preventDefault}>
              Already have an account? Login here
            </Link>
          </Typography>
        </form>
      </div>
    );
  }
}

export default Register;
