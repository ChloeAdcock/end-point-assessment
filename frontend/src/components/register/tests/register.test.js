import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import Register from '../Register';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe("The register component", () => {
    it("should render the register form when initial state", () => {
      const store = mockStore({
        accounts: {
          currentUser: null,
          currentUserId: null,
          registerError: null,
        },
      });
      const tree = renderer
        .create(
          <Provider store={store}>
              <Register />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it("should render an error alert when registerError is true", () => {
      const store = mockStore({
        accounts: {
          currentUser: null,
          currentUserId: null,
          registerError: true,
        },
      });
      const tree = renderer
        .create(
          <Provider store={store}>
              <Register />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });