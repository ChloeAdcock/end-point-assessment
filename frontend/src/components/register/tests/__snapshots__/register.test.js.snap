// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The register component should render an error alert when registerError is true 1`] = `
<div>
  <div
    className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
    role="alert"
  >
    <div
      className="MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message"
    >
      An error has occurred - please try again later
    </div>
    <div
      className="MuiAlert-action"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Close"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
      varient="h1"
    >
      Register
    </p>
    <div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            aria-describedby={null}
            aria-label="info"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Username must be alphanumeric and 5-10 characters"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
            >
              Username
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Username
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
      >
        Email
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Email
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            aria-describedby={null}
            aria-label="info"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Password must be alphanumeric, 5-10 characters and contain at least one letter and one number"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
            >
              Password
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Password
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
      >
        Confirm password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Confirm password
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Submit
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="/login"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        Already have an account? Login here
      </a>
    </p>
  </form>
</div>
`;

exports[`The register component should render the register form when initial state 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
      varient="h1"
    >
      Register
    </p>
    <div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            aria-describedby={null}
            aria-label="info"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Username must be alphanumeric and 5-10 characters"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
            >
              Username
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Username
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
      >
        Email
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Email
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            aria-describedby={null}
            aria-label="info"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Password must be alphanumeric, 5-10 characters and contain at least one letter and one number"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
            >
              Password
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Password
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
      >
        Confirm password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Confirm password
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Submit
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="/login"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        Already have an account? Login here
      </a>
    </p>
  </form>
</div>
`;
